controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1755,18 +1795,31 @@ func (r *BareMetalHostReconciler) createHostFirmwareComponents(info *reconcileIn
 
 			// Set bmh as owner, this makes sure the resource is deleted when bmh is deleted
 			if err = controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
```

Comment:
Controller references are exclusive, and almost never what you want. Is there a reason we are doing this instead of a regular owner reference?

Reply:
An HFC object is tightly coupled with a BareMetalHost (it's the same Ironic Node), so I think it's justified.

Reply:
Controller references aren't a value judgement on how tightly coupled objects are, they're to solve a specific problem between ReplicaSet and Pod that doesn't exist here.
See https://github.com/kubernetes/design-proposals-archive/blob/acc25e14ca83dfda4f66d8cb1f1b491f26e78ffe/api-machinery/controller-ref.md